pkg/providers/instance/types.go

```diff
@@ -40,7 +42,7 @@ type Instance struct {
 	EFAEnabled            bool
 }
 
-func NewInstance(out ec2types.Instance) *Instance {
+func NewInstance(ctx context.Context, out ec2types.Instance) *Instance {
```

Comment:
nit: Maybe worth if we note down why we are using this methodology to block returning back reserved capacity type